ledger/internal/cow.go

```diff
@@ -118,21 +122,24 @@ func (cb *roundCowState) deltas() ledgercore.StateDelta {
 	//    SetKey/DelKey work only with sdeltas, so need to pull missing accounts
 	// 2. Call applyStorageDelta for every delta per account
 	for addr, smap := range cb.sdeltas {
-		var delta basics.AccountData
-		var exist bool
-		if delta, exist = cb.mods.Accts.Get(addr); !exist {
-			ad, err := cb.lookup(addr)
+		for aapp, storeDelta := range smap {
+			// app params and app local states might be accessed without touching base account record
+			// from TEAL by writing/deleting KV store
+			// so store the base account to deltas here
+			// TODO: remove after the schema switch
+			acct, err := cb.lookup(addr)
```

Comment:
this method need to be re-written so that instead of panic, we would return an error. This doesn't need to be done in the context of this PR.